kai/service/llm_interfacing/model_provider.py

```diff
@@ -51,9 +52,14 @@ def __init__(self, config: KaiConfigModels):
 
             case "ChatIBMGenAI":
                 model_class = LangChainChatInterface
-
+                if os.getenv("DEMO_MODE") == "true":
+                    credentials = Credentials(
+                        api_key="GENAI_KEY", api_endpoint="dummy_endpoint"
```

Comment:
@Myrausman let's look to see if a value has been set for GENAI_KEY and/or GENAI_API and use those values IF they exist.  If they do not exist then use the values you have.

The reason for this is that with DEMO_MODE=True, while we do want to play back cached responses, we also sometimes have the need to issue real requests for responses that were not previously cached.  For that case the real values would then need to be present.  


Reference info:
GENAI_KEY=YOUR_GENAI_API_KEY
GENAI_API=https://bam-api.res.ibm.com

https://ibm.github.io/ibm-generative-ai/v3.0.0/getting_started.html#id5

Reply:
Hi @jwmatthews, I've updated the code as you suggested. Now, the logic checks if GENAI_KEY and GENAI_API environment variables are set. If they are, those values will be used; otherwise, the default values will be applied.